Jetpack Section: Restore Status and Restore Complete

momo-ozawa opened the pull request:
Part of #15191

- Hooked up restore site APIs
    - Restore site
    - Get rewind status (poll for status **every 5 seconds**)
- Added a new Restore Complete screen

### Merge directions:
- ⚠️ Merge https://github.com/wordpress-mobile/WordPress-iOS/pull/15633 first

### Notes:
- Styling isn't 100% complete
- The `Visit site` button on the Restore Complete screen isn't hooked up yet
- Error handling will be addressed in a separate PR

### To test:

1. My Site > Activity Log > rewindable activity cell
2. Tap on ellipsis icon
3. Tap on `Restore`
4. Tap on `Restore to this point`
5. Tap on `Confirm`
     - ✅ The progress view should update
     - ✅ Once the site has been restored, the app should transition to the Restore Complete screen (Note that it might take a little while to transition after it hits `100%`)
     

30% (`.running`)  | 100% (`.running`) | Complete (`.finished`)
--- | --- | ---
![Screen Shot 2021-01-14 at 19 38 26](https://user-images.githubusercontent.com/6711616/104580236-57a32d80-56a0-11eb-8ad9-17c7771312db.png) | ![Screen Shot 2021-01-14 at 19 38 34](https://user-images.githubusercontent.com/6711616/104580247-5a058780-56a0-11eb-88b5-d19a5130f71f.png) | ![Screen Shot 2021-01-14 at 19 38 55](https://user-images.githubusercontent.com/6711616/104580251-5c67e180-56a0-11eb-8285-38aa93c96662.png) 

### PR submission checklist:

- [x] I have considered adding unit tests where possible.
- [x] I have considered adding accessibility improvements for my changes.
- [ ] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

Commit:
Add Jetpack Backup Status Coordinator

Commit:
Add Jetpack Restore Service

Commit:
Configure Jetpack Restore Status Coordinator

Commit:
Add Jetpack Restore Complete VC

Commit:
Override button handlers in subclasses

leandroalonso commented on WordPress/Classes/ViewRelated/Jetpack/Jetpack Restore/Restore Complete/JetpackRestoreCompleteViewController.swift:
```diff
@@ -0,0 +1,43 @@
+            title: NSLocalizedString("Restore", comment: "Title for Jetpack Restore Complete screen"),
+            iconImage: .gridicon(.history),
+            messageTitle: NSLocalizedString("Your site has been restored", comment: "Title for the Jetpack Restore Complete message."),
+            messageDescription: NSLocalizedString("All of your selected itmes are now restored back to %1$@.", comment: "Description for the Jetpack Backup Restore message. %1$@ is a placeholder for the selected date."),
```
```suggestion
            messageDescription: NSLocalizedString("All of your selected items are now restored back to %1$@.", comment: "Description for the Jetpack Backup Restore message. %1$@ is a placeholder for the selected date."),
```

Typo: itmes

momo-ozawa replied:
Fixed! 24c9a6e

leandroalonso commented on WordPress/Classes/Services/JetpackRestoreService.swift:
```diff
@@ -0,0 +1,34 @@
+@objc class JetpackRestoreService: LocalCoreDataService {
+
+    private lazy var serviceV1: ActivityServiceRemote_ApiVersion1_0 = {
+        let api = WordPressComRestApi.defaultApi(in: managedObjectContext,
```
You can omit the `localeKey`

momo-ozawa replied:
Fixed! 10cd307

leandroalonso reviewed (commented):
@momo-ozawa works as described!

One question though, the confirmation message says: 

<img width="372" alt="Screen Shot 2021-01-14 at 12 56 29" src="https://user-images.githubusercontent.com/7040243/104615366-f1021d80-5667-11eb-981e-8152e88af2a4.png">

However, this might be not true: what if the user selected just WordPress Themes?

cc @osullivanchris

Commit:
Fix typo

Commit:
Delete localeKey parameter from call

momo-ozawa commented:
@leandroalonso @osullivanchris 

> However, this might be not true: what if the user selected just WordPress Themes?

Hmm, true. This is what Wordpress.com looks like. 

![Screen Shot 2021-01-15 at 10 35 02](https://user-images.githubusercontent.com/6711616/104670766-cc697c80-571f-11eb-88e7-f2a197e458ef.png)

osullivanchris commented:
> However, this might be not true: what if the user selected just WordPress Themes?

Good spot. I pulled the message from something already existing on web. But maybe I didn't grab the right one.

@momo-ozawa in that screenshot, does the message change depending on what you select? 

Either way, my intention was to follow what we are doing on web. So if we have something like this already, that's great, let's use it. If we don't, then we can come up with our own alternative.

osullivanchris commented:
Discussed this [on Android over here](https://github.com/wordpress-mobile/WordPress-Android/pull/13775#issuecomment-760960333). After discussion with @ashiagr - realised that the word 'All' is really the only one causing issue. 

If we remove the word 'all', the sentence still works grammatically, but it doesn't conclusively say that we are removing *everything* when the user has only selected *some* things. That should resolve the issue I think.

cc @leandroalonso

Commit:
Merge branch 'develop' into task/15191-restore-status

Commit:
Update warning description format

Commit:
Disable action button if there are no items selected

momo-ozawa commented:
@leandroalonso (cc: @osullivanchris) 

- Removed the word "all"
- Restore action is disabled if no items are selected

Updated restore warning copy | No items selected | At least 1 item selected 
--- | --- | ---
![Simulator Screen Shot - iPhone 12 Pro - 2021-01-16 at 10 39 45](https://user-images.githubusercontent.com/6711616/104793792-3b5dd880-57e7-11eb-8b7f-c4d2c905beae.png) | ![Simulator Screen Shot - iPhone 12 Pro - 2021-01-16 at 10 39 56](https://user-images.githubusercontent.com/6711616/104793796-3c8f0580-57e7-11eb-81c5-5aa7e03ea011.png) | ![Simulator Screen Shot - iPhone 12 Pro - 2021-01-16 at 10 40 01](https://user-images.githubusercontent.com/6711616/104793798-3d279c00-57e7-11eb-926a-cccd9c7f9732.png)
